### PR TITLE
Merging testing branch

### PR DESCRIPTION
**Pull request by TkTech**
- circle.ci runners for postgresql and sqlite3.
- Fix an issue with account activation introduced by #111
- Backport upstream fixes for PreviousLogin logic, PreviousLogin test cases.
- Check if a domain is valid before bothering with a an expensive request to elgg to look for invites.
- Update settings to:
   - Default to sqlite3 as a database option.
   - Use `dj_database_url` to configure the database from an environ
   - Load `ALLOWED_HOSTS`, `DATABASES`, `SECRET_KEY` from environ
   - Update docker config to reflect these changes

### CODE REVIEW

File: core/models.py [hunk 1]
```diff
@@ -148,39 +149,32 @@ def activate_user(self, activation_token):
...
+            )
             login.update_previous_login(request)
-
-        if send_suspicious_behavior_warnings:
-            if not self.logged_in_previously(request):
-                # no confirmed matching login found, so email must be sent
+            return True
+        except PreviousLogin.DoesNotExist:
+            PreviousLogin.add_known_login(request, self)
+            if self.receives_newsletter is True:
```
Comment (bryan-robitaille) [inline core/models.py]: At this point we might as well just refactor everything in the app for 'receives_newsletter' to 'send_suspicious_login'.  It's the only thing the bool is used for.  Doesn't have to be done now, but if you're up for it?

File: core/models.py [hunk 2]
```diff
@@ -148,39 +149,32 @@ def activate_user(self, activation_token):
 
             self.is_active = True
             self.save()
-            #valid_user is the routing followed by name, email and id
-            data = json.dumps({'name': self.name, 'email':self.email, 'id': self.id })
-            routing = 'user.new'
-            mq_newuser(routing,data)
+            # valid_user is the routing followed by name, email and id
+            mq_newuser('user.new', json.dumps({
```
Comment (bryan-robitaille) [inline core/models.py]: Thanks, much cleaner

File: pleio_account/settings.py [hunk 3]
```diff
@@ -350,3 +365,20 @@
     ))
 ])
 
+SECRET_KEY = os.environ.get('CONCIERGE_SECRET_KEY')
+
+default_db_path = os.path.join(BASE_DIR, 'db.sqlite3')
+DATABASES = {
+    'default': dj_database_url.config(
```
Comment (bryan-robitaille) [inline pleio_account/settings.py]: ok, that's cool and really simplifies things.  Didn't know about dj_database_url package.

File: core/models.py (hunk 1 above)
Comment (TkTech) [reply core/models.py]: Makes sense, will do.

File: pleio_account/settings.py [hunk 4]
```diff
@@ -118,28 +121,37 @@
 
 SESSION_ENGINE = 'user_sessions.backends.db'
 
-# To be replaced later with REDIS when merging with Account Lockout
 CACHES = {
     'default': {
-        'BACKEND': 'django.core.cache.backends.locmem.LocMemCache',
-        'LOCATION': 'default-cache',
+        'BACKEND': 'redis_cache.RedisCache',
```
Comment (bryan-robitaille) [inline pleio_account/settings.py]: This redis library is not part of the django-redis package in requirements.txt but instead use django-redis-cache.  Please update either the backend library called or the package requirements

File: pleio_account/settings.py [hunk 5]
```diff
@@ -214,11 +227,18 @@
...
 MQ_PASSWORD = ""
 MQ_CONNECTION = ""
 
+# django-defender
+DEFENDER_LOGIN_FAILURE_LIMIT = 5
+DEFENDER_DISABLE_IP_LOCKOUT = True
+DEFENDER_COOLOFF_TIME = 600
+DEFENDER_ACCESS_ATTEMPT_EXPIRATION = 24
+DEFENDER_USERNAME_FORM_FIELD = 'auth-username'
+
```
Comment (bryan-robitaille) [inline pleio_account/settings.py]: Defender requires a missing variable DEFENDER_REDIS_URL or DEFENDER_REDIS_NAME to direct it to the redis instance.

File: pleio_account/settings.py [hunk 6]
```diff
@@ -350,3 +374,20 @@
     ))
 ])
 
+SECRET_KEY = os.environ.get('CONCIERGE_SECRET_KEY')
+
```
Comment (bryan-robitaille) [inline pleio_account/settings.py]: Given that there are defaults for database and allowed hosts is it also possible to have a default for the secret key in a development environment?  Maybe if you're using the SQLite DB then secret key can be auto set?

File: pleio_account/settings.py (hunk 5 above)
Comment (TkTech) [reply pleio_account/settings.py]: Good point, I was just running it on localhost so the default worked-for-me :) Will add the settings from envvar.

**Comment by bryan-robitaille** [comment]
Merged into master branch with #118

**Pull request closed without merging**